src/app/boot/app_controller.nim

```diff
@@ -310,6 +312,47 @@ proc delete*(self: AppController) =
   self.gifService.delete
   self.keycardService.delete
 
+proc disconnectKeychain(self: AppController) =
+  for id in self.keychainConnectionIds:
+    self.statusFoundation.events.disconnect(id)
+  self.keychainConnectionIds = @[]
+
+proc connectKeychain(self: AppController) =
+  var handlerId = self.statusFoundation.events.onWithUUID(SIGNAL_KEYCHAIN_SERVICE_SUCCESS) do(e:Args):
+    let args = KeyChainServiceArg(e)
+    self.disconnectKeychain()
+    ## we need to set local `storeToKeychain` prop to `store` value since in this context means pass/pin is stored well
+    singletonInstance.localAccountSettings.setStoreToKeychainValue(LS_VALUE_STORE)
+  self.keychainConnectionIds.add(handlerId)
+
+  handlerId = self.statusFoundation.events.onWithUUID(SIGNAL_KEYCHAIN_SERVICE_ERROR) do(e:Args):
+    let args = KeyChainServiceArg(e)
+    self.disconnectKeychain()
+    ## no need for any other activity in this context, local `storeToKeychain` prop remains as it was
+    ## maybe in some point in future we add a popup letting user know about this
+    info "unable to store the data to keychain", errCode=args.errCode, errType=args.errType, errDesc=args.errDescription
+  self.keychainConnectionIds.add(handlerId)
+
+proc checkForStoringPasswordToKeychain(self: AppController) =
+  ## This proc is used to store pass/pin depends on user's selection during onboarding flow.
+  let account = self.accountsService.getLoggedInAccount()
+  let value = singletonInstance.localAccountSettings.getStoreToKeychainValue()
+  if not main_constants.IS_MACOS or # This is MacOS only feature
```

Comment:
This looks like we don't plan to add fingerprints to other OSes ;)
Having a `SUPPORTS_FINGERPRINT` state instead of `IS_MACOS` would improve things in this regard.

Reply:
> const SUPPORTS_FINGERPRINT = if MACOS: return true else: return false

Exactly that sounds good to me. I'm thinking this also documents the condition enough so you can remove the verbose comments.